librocksdb_sys/crocksdb/c.cc

```diff
@@ -2853,4 +2855,49 @@ crocksdb_logger_t *crocksdb_create_log_from_options(const char *path,
 
 void crocksdb_log_destroy(crocksdb_logger_t *logger) { delete logger; }
 
+crocksdb_pinnableslice_t* crocksdb_get_pinned(
+    crocksdb_t* db, const crocksdb_readoptions_t* options, const char* key,
+    size_t keylen, char** errptr) {
+  crocksdb_pinnableslice_t* v = new (crocksdb_pinnableslice_t);
+  Status s = db->rep->Get(options->rep, db->rep->DefaultColumnFamily(),
+                          Slice(key, keylen), &v->rep);
+  if (!s.ok()) {
+    delete (v);
+    if (!s.IsNotFound()) {
+      SaveError(errptr, s);
+    }
+    return NULL;
+  }
+  return v;
+}
+
+crocksdb_pinnableslice_t* crocksdb_get_pinned_cf(
+    crocksdb_t* db, const crocksdb_readoptions_t* options,
+    crocksdb_column_family_handle_t* column_family, const char* key,
+    size_t keylen, char** errptr) {
+  crocksdb_pinnableslice_t* v = new (crocksdb_pinnableslice_t);
+  Status s = db->rep->Get(options->rep, column_family->rep, Slice(key, keylen),
+                          &v->rep);
+  if (!s.ok()) {
+    delete v;
+    if (!s.IsNotFound()) {
+      SaveError(errptr, s);
+    }
+    return NULL;
+  }
+  return v;
+}
+
+void crocksdb_pinnableslice_destroy(crocksdb_pinnableslice_t* v) { delete v; }
+
+const char* crocksdb_pinnableslice_value(const crocksdb_pinnableslice_t* v,
+                                        size_t* vlen) {
+  if (!v) {
```

Comment:
In our case, when call `crocksdb_pinnableslice_value` the `v` is not null, but there is no guarantee that `v` is not null in other cases.